Sources/AblyChat/ChatAPI.swift

```diff
@@ -0,0 +1,232 @@
+            headers: params.headers ?? [:]
+        )
+        return message
+    }
+
+    func getOccupancy(roomId: String) async throws -> OccupancyEvent {
+        let endpoint = "/chat/v1/rooms/\(roomId)/occupancy"
+        return try await makeAuthorizedRequest(endpoint, method: "GET")
+    }
+    
+    private func makeAuthorizedRequest<RES: Codable>(_ url: String, method: String, body: [String: Any]? = nil) async throws -> RES {
+            return try await withCheckedThrowingContinuation { continuation in
+                do {
+                    try rest.request(method, path: url, params: [:], body: body, headers: ["protocol": String(apiProtocolVersion)], callback: { paginatedResponse, error  in
+                        
+                        if let error {
+                            continuation.resume(throwing: error)
+                            return
+                        }
+                        
+                        guard let firstItem = paginatedResponse?.items.first else {
+                            continuation.resume(throwing: ARTErrorInfo.createUnknownError())
+                            return
+                        }
+                        
+                        do {
+                            let jsonData = try JSONSerialization.data(withJSONObject: firstItem)
+                            let decodedResponse = try JSONDecoder().decode(RES.self, from: jsonData)
+                            continuation.resume(returning: decodedResponse)
+                        } catch {
+                            continuation.resume(throwing: error)
+                        }
+                    })
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+    }
+    
+    private func makeAuthorizedPaginatedRequest<RES: Codable & Sendable>(
+        _ url: String,
+        params: [String: String]? = nil,
+        body: [String: Any]? = nil
+    ) async throws -> any PaginatedResult<RES> {
+        if #available(macOS 13.0.0, *) {
+            return try await withCheckedThrowingContinuation { continuation in
+                do {
+                    try rest.request("GET", path: url, params: params, body: nil, headers: ["protocol": String(apiProtocolVersion)], callback: { paginatedResponse, error  in
+                        
+                        ARTHTTPPaginatedCallbackWrapper<RES>(callback: (paginatedResponse, error)).handleResponse(continuation: continuation)
+                    })
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+            
+        } else {
+            // Fallback on earlier versions
+            throw ARTErrorInfo.create(withCode: 501, status: 0, message: "Unsupported macOS version")
+        }
+    }
+}
+
+@available(macOS 13.0.0, *)
+struct ARTHTTPPaginatedCallbackWrapper<RES: Codable & Sendable> {
```

Comment:
It's not obvious what this class does; please could you add some documentation?

Reply:
added